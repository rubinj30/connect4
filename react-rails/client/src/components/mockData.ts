import { PieceType } from './atoms/space.component';
import { ColumnType } from './molecules/column.component';
export const altColumn = ['B', 'R', 'B', ' ', ' ', ' '];
export const winColumn: PieceType[] = ['B', 'B', 'B', 'B', ' ', ' '];
export const blankColumn: PieceType[] = [' ', ' ', ' ', ' ', ' ', ' '];
export const fullColumn: PieceType[] = ['B', 'R', 'B', 'B', 'R', 'R'];
export const blankBoard: ColumnType[] = [
    blankColumn,
    blankColumn,
    blankColumn,
    blankColumn,
    blankColumn,
    blankColumn,
    blankColumn
];

export const nonWinBoard = [
    altColumn,
    blankColumn,
    altColumn,
    blankColumn,
    altColumn,
    blankColumn,
    altColumn
];

export const someColsFull = [
    fullColumn,
    fullColumn,
    blankColumn,
    blankColumn,
    blankColumn,
    blankColumn,
    blankColumn
];
