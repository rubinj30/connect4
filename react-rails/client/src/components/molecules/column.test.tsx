jest.unmock('./column.component');

import React from 'react';
import { shallow, mount } from 'enzyme';
import { Column } from './column.component';
import { winColumn, fullColumn } from '../mockData';

describe('Column', () => {
    let mockProps;
    beforeEach(() => {
        mockProps = {
            column: winColumn,
            win: false,
            dataIndex: 1,
            currentTurn: 'B',
            isCompTurn: 'n',
            handleClick: jest.fn()
        };
    });
    it('should render proprerly', () => {
        const fixture = <Column {...mockProps} />;
        const result = shallow(fixture);
        expect(result).toBeDefined();
        expect(result).toMatchSnapshot();
    });
    it('anyBlankSpaces should return true if there are blank spaces in the provided column', () => {
        const fixture = <Column {...mockProps} />;
        const result = shallow(fixture);
        const instance = result.instance();
        const anyBlanks = instance.anyBlankSpaces(winColumn);
        expect(anyBlanks).toBeTruthy();
    });
    it('anyBlankSpaces should return false if there are no spaces in the provided column', () => {
        const fixture = <Column {...mockProps} />;
        const result = shallow(fixture);
        const instance = result.instance();
        const anyBlanks = instance.anyBlankSpaces(fullColumn);
        expect(anyBlanks).toBeFalsy();
    });
    it('anyBlankSpaces should allow handleClick to be called if there are blank spaces in a column', () => {
        const fixture = <Column {...mockProps} />;
        const result = mount(fixture);
        const instance = result.instance();
        result.childAt(0).simulate('click');
        console.log(result.childAt(0).debug())
        expect(result.props().handleClick).toHaveBeenCalledTimes(1);
    });
});
