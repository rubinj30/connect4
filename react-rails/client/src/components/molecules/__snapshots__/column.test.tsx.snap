// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Column should render proprerly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Column
    column={
      Array [
        "B",
        "B",
        "B",
        "B",
        " ",
        " ",
      ]
    }
    currentTurn="B"
    dataIndex={1}
    handleClick={[MockFunction]}
    isCompTurn="n"
    win={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Arrow
          currentTurn="B"
          win={false}
        />,
        Array [
          <Space
            piece=" "
          />,
          <Space
            piece=" "
          />,
          <Space
            piece="B"
          />,
          <Space
            piece="B"
          />,
          <Space
            piece="B"
          />,
          <Space
            piece="B"
          />,
        ],
      ],
      "className": "flex flex-column column justify-between items-center",
      "data-index": 1,
      "onClick": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "currentTurn": "B",
          "win": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "5",
        "nodeType": "class",
        "props": Object {
          "piece": " ",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "4",
        "nodeType": "class",
        "props": Object {
          "piece": " ",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "class",
        "props": Object {
          "piece": "B",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "class",
        "props": Object {
          "piece": "B",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "piece": "B",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "class",
        "props": Object {
          "piece": "B",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Arrow
            currentTurn="B"
            win={false}
          />,
          Array [
            <Space
              piece=" "
            />,
            <Space
              piece=" "
            />,
            <Space
              piece="B"
            />,
            <Space
              piece="B"
            />,
            <Space
              piece="B"
            />,
            <Space
              piece="B"
            />,
          ],
        ],
        "className": "flex flex-column column justify-between items-center",
        "data-index": 1,
        "onClick": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "currentTurn": "B",
            "win": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "5",
          "nodeType": "class",
          "props": Object {
            "piece": " ",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "class",
          "props": Object {
            "piece": " ",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "class",
          "props": Object {
            "piece": "B",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "class",
          "props": Object {
            "piece": "B",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "piece": "B",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "piece": "B",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
